src/polyswarm_api/__init__.py

```diff
@@ -310,67 +313,84 @@ def _fix_result(self, result):
         logger.warning('Failed to get results for uuid %s in time.', uuid)
         return {'files': result['files'], 'uuid': uuid}
 
-    async def scan_url(self, to_scan, name='url'):
+    async def scan_fileobjs(self, file_objs, file_names, artifact_type):
         """
-        Scan a single URL using the PS API asynchronously
+        Scan a collection of file-like object using the PS API asynchronously.
 
-        :param to_scan: URL to scan
-        :param name: name to associate with the artifact
+        :param to_scan: A list of file-like objects to scan.
+        :param filenames: A list of filenames to use
         :return: JSON report
         """
-        result = await self.post_url(to_scan, name)
-        if result['status'] == 'OK':
-            uuid = result['result']
-        else:
-            logger.error('Failed to get UUID for scan of file %s', name)
-            return {'filename': name, 'files': []}
-
-        logger.info('Successfully submitted file %s, UUID %s', name, uuid)
+        futures = []
+        file_objs = grouper(MAX_ARTIFACT_BATCH_SIZE, file_objs)
+        file_names = grouper(MAX_ARTIFACT_BATCH_SIZE, file_names)
+        for artifacts, names in zip(file_objs, file_names):
+            futures.append(self.post_artifacts(artifacts, names, artifact_type=artifact_type))
+        return await asyncio.gather(*futures)
 
-        return await self._wait_for_uuid(uuid)
+    async def wait_for_results(self, results):
+        futures = []
+        for result in results:
+            futures.append(self._wait_for_uuid(result['result']))
+        return await asyncio.gather(*futures)
 
-    async def scan_fileobj(self, to_scan, filename='data'):
+    async def scan_urls(self, to_scan):
         """
-        Scan a single file-like object using the PS API asynchronously.
+        Scan a single URL using the PS API asynchronously
 
-        :param to_scan: File-like object to scan.
-        :param filename: Filename to use
+        :param to_scan: List of URLs to scan
         :return: JSON report
         """
+        try:
+            file_objs = [io.StringIO(url) for url in to_scan]
+            file_names = ['url'] * len(to_scan)
+            result = await self.scan_fileobjs(file_objs, file_names, artifact_type=ArtifactType.URL)
+            return await self.wait_for_results(result)
+        except exceptions.PolyswarmAPIException as e:
```

Comment:
Add `status: 'error'` and `result: 'error'` to this exception dict to make things consistent for the formatter.

Reply:
Fixed :+1: